app/controllers/time_logs_controller.rb

```diff
@@ -0,0 +1,50 @@
+class TimeLogsController < ApplicationController
+  before_action :set_project, only: [ :create, :new ]
+  before_action :set_time_log, only: [ :edit, :update, :destroy ]
+  before_action :validate_user, only: [ :edit, :update, :destroy ]
+
+  def index
+    @time_logs = TimeLog.includes(:user, :project).all
+  end
+
+  def new
+    @time_log = TimeLog.new
+  end
+
+  def create
+    @time_log = TimeLog.new(time_log_params)
+    @time_log.project = @project
```

Comment:
again this is a new one.
do you think there could be another way to do it, you have used it in other controllers also.